plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -528,7 +530,12 @@ public MongoCursor<Document> execute(MongoTableHandle tableHandle, List<MongoCol
 
         MongoCollection<Document> collection = getCollection(tableHandle.remoteTableName());
         Document filter = buildFilter(tableHandle);
-        FindIterable<Document> iterable = collection.find(filter).projection(projection).collation(SIMPLE_COLLATION);
+        FindIterable<Document> iterable = collection.find(filter).projection(projection);
+
+        if (!skipCollation) {
```

Comment:
MongoDB doesn't expose whether collation is supported or not, right? I'm asking because we'd like to avoid using configuration properties as much as possible. Ideally, this skip should be determined automatically rather than relying on a flag.

Reply:
@ebyhr I tried searching quickly for that and it doesn't seem to be the case that collation is a capability you can simply query for its presence or not .. as far as I have read the only way would be to attempt a try / catch query with collation to see if it's supported or not .. which in my view is not really an elegant solution